src/insertItem.ts

```diff
@@ -0,0 +1,482 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+// tslint:disable-next-line:no-duplicate-imports
+import { commands } from "vscode";
+import { IAzureUserInput } from "vscode-azureextensionui";
+import { Json, templateKeys } from "../extension.bundle";
+import { DeploymentTemplate } from "./DeploymentTemplate";
+import { ext } from './extensionVariables';
+import { SortType } from "./sortTemplate";
+
+const insertCursorText = '[]';
+
+export class QuickPickItem<T> implements vscode.QuickPickItem {
+    public label: string;
+    public value: T;
+    public description: string;
+
+    constructor(label: string, value: T, description: string) {
+        this.label = label;
+        this.value = value;
+        this.description = description;
+    }
+}
+
+export function getItemType(): QuickPickItem<string>[] {
+    let items: QuickPickItem<string>[] = [];
+    items.push(new QuickPickItem<string>("String", "string", "A string"));
+    items.push(new QuickPickItem<string>("Secure string", "securestring", "A secure string"));
+    items.push(new QuickPickItem<string>("Int", "int", "An integer"));
+    items.push(new QuickPickItem<string>("Bool", "bool", "A boolean"));
+    items.push(new QuickPickItem<string>("Object", "object", "An object"));
+    items.push(new QuickPickItem<string>("Secure object", "secureobject", "A secure object"));
+    items.push(new QuickPickItem<string>("Array", "array", "An array"));
+    return items;
+}
+
+export function getResourceSnippets(): vscode.QuickPickItem[] {
+    let items: vscode.QuickPickItem[] = [];
```

Comment:
Use assetsPath from constants.ts and read in armSnippets.json as a JSON file to get the list of snippets. For the moment I think you can search the snippet contents and if you find "Microsoft." in the snippet you can assume it's a resource snippet.  Eventually I'm going to put these more directly under our control (so users only see snippets in the right places), but this will work fine for now.

Reply:
Resource snippets are now read from armsnippets.jsonc as you suggested.